src/tests/suite/daos_nvme_recovery.c

```diff
@@ -119,9 +119,174 @@ nvme_recov_1(void **state)
 	print_message("Done\n");
 }
 
+/* Verify device states after NVMe set to faulty*/
+static void
+nvme_test_verify_device_stats(void **state)
+{
+	test_arg_t	*arg = *state;
+	device_list	*devices = NULL;
+	daos_obj_id_t	oid;
+	char		data_buf[100];
+	char		fetch_buf[100] = { 0 };
+	struct ioreq	req;
+	int		ndisks;
+	int		rc, i;
+	char		*server_config_file;
+	char		*log_file;
+	int		obj_class;
+
+	if (!is_nvme_enabled(arg)) {
+		print_message("NVMe isn't enabled.\n");
+		skip();
+	}
+
+	/**
+	*Get the Total number of NVMe devices from all the servers.
+	*/
+	rc = dmg_storage_device_list(dmg_config_file, &ndisks, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Total Disks = %d\n", ndisks);
+
+	/**
+	*Get the Device info of all NVMe devices.
+	*/
+	D_ALLOC_ARRAY(devices, ndisks);
+	rc = dmg_storage_device_list(dmg_config_file, NULL, devices);
+	assert_int_equal(rc, 0);
+	for (i = 0; i < ndisks; i++)
+		print_message("Rank=%d UUID=%s state=%s host=%s\n",
+			      devices[i].rank, DP_UUID(devices[i].device_id),
+			devices[i].state, devices[i].host);
+
+	/*
+	 * Get the server config file from running process on server.
+	 * Verify log_mask in server.yaml file, It should be 'DEBUG' to verify
+	 * different state of NVMe drives. Skip the test if log_mask is not
+	 * set to DEBUG.
+	 */
+	D_ALLOC(server_config_file, 512);
+	D_ALLOC(log_file, 1024);
+	for (i = 0; i < ndisks; i++) {
+		if (devices[i].rank == 1) {
+			rc = get_server_config(devices[i].host,
+					       server_config_file);
+			assert_int_equal(rc, 0);
+			print_message("server_config_file = %s\n",
+				      server_config_file);
+
+			get_server_log_file(devices[i].host,
+					    server_config_file, log_file);
+			rc = verify_server_log_mask(devices[i].host,
+						    server_config_file,
+						    "DEBUG");
+			if (rc) {
+				print_message("Log Mask != DEBUG in %s.\n",
+					      server_config_file);
+				skip();
+			}
+		}
+	}
+	print_message("LOG FILE = %s\n", log_file);
+
+	/** Prepare records **/
+	if (arg->pool.pool_info.pi_nnodes < 2)
+		obj_class = DAOS_OC_R1S_SPEC_RANK;
+	else
+		obj_class = DAOS_OC_R2S_SPEC_RANK;
+
+	oid = dts_oid_gen(obj_class, 0, arg->myrank);
+	oid = dts_oid_set_rank(oid, 1);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
```

Comment:
missing ioreq_fini() ?

Reply:
Yes let me add it, Thanks for catching it.